packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -355,18 +310,30 @@ function parseChildren(
     parse5ParentLocation: parse5.ElementLocation
 ): void {
     const children = (parse5Utils.getTemplateContent(parse5Parent) ?? parse5Parent).childNodes;
+
+    let siblingContext: IfBlock | undefined;
```

Comment:
This is the loop where we iterate through the siblings of a particular node and where can keep track of any context we need from sibling to sibling.